### PR TITLE
FC-0068 docs: Documentation Templates (How-to, Reference, Concept)

### PR DESCRIPTION
**Pull request by Apgomeznext**
This PR adds templates for the different document types according to the diataxis used in the Open edX repository.

Review version: https://docsopenedxorg--619.org.readthedocs.build/en/619/documentors/references/doc_templates.html

How To template doc: https://docs.google.com/document/d/1nDCoSnNApQciYD042cqCadk94LzWf4kQSPsGi0VeIBI/edit?tab=t.0#heading=h.fsh8gw4ee3kc
Reference template: https://docs.google.com/document/d/1QHr5sfKeeQ2_PzOMfw53O8QOZHJoxxNyKcuFhQUHCDw/edit?tab=t.0#heading=h.6b7r0b9noosl
Concept Template: https://docs.google.com/document/d/1n3hfDkPl2ftsPreLz601Aa_zh4E6Zqft6d8ZcMCAZK0/edit?tab=t.0#heading=h.6b7r0b9noosl

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Apgomeznext!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/docs-openedx-org-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: source/documentors/references/doc_templates.rst
Comment (jswope00) [inline source/documentors/references/doc_templates.rst:1]: This seems redundant now that we have source/documentors/references/templates/index.rst. 

The error you are getting seems somewhat related to the fact that this doc_templates.rst file exists, but it is not in any TOC.

**Review by jswope00 — CHANGES_REQUESTED** [review]
The changes removed a reference to doc_templates.rst file, which is causing a "does not exist in TOC" error. It looks like that file can be deleted because it has been replaced by index.rst.

**Review by jswope00 — APPROVED** [review]
Looks good

File: source/documentors/references/templates/doc_concept_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_concept_template.rst:1]: @Apgomeznext there is a punctuation error (last full stop) which should be fixed, and also there is an extra space between brackets:
Lines 6 should be like this: For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.” (Replace this text with your content).

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_how_to_template.rst:6]: Add a full stop after "(Description)."

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_how_to_template.rst:51]: Same here with punctuation.

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_quickstart_template.rst:60]: Add a final stop after "Quickstart".

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_quickstart_template.rst:78]: Add a final stop after "Quickstart".

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_quickstart_template.rst:112]: Should we add the link to the Style Guide, right?

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_quickstart_template.rst:179]: Final stop after "Quickstart".

File: source/documentors/references/templates/doc_reference_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_reference_template.rst:6]: Full stop should be after the parenthesis:  (Replace this text with your content).

File: source/documentors/references/templates/doc_reference_template.rst
Comment (jesgreco) [inline source/documentors/references/templates/doc_reference_template.rst:79]: Full stop should be after the parenthesis: (Replace this text with your content).

**Review by jesgreco — COMMENTED** [review]
I've left some comments mostly related to punctuation to be fixed :)

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:6]:
```suggestion
For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.”  (Replace this text with your content).
```

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:8]:
```suggestion
Since this type of document seeks to explain a topic, you will likely need to add images to illustrate the topic you are talking about. For example:
```

"exemplify" is a more complicated english word than "illustrate"

File: source/documentors/references/templates/index.rst
Comment (sarina) [inline source/documentors/references/templates/index.rst:1]:
```suggestion
Open edX Document Templates
```

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:11]: Why is the example text above about XBlocks, but this image unrelated to XBlocks? It's confusing. I'd make the image and the example text above at line 6 match.

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:19]:
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_reference_template.rst
Comment (sarina) [reply source/documentors/references/templates/doc_reference_template.rst:6]:
```suggestion
Reference material is like a map. A map tells you what you need to know about the territory without having to go out and check it for yourself; a reference guide serves the same purpose for the product and its internal machinery. For example, this part works as the introduction to the article and as a description of the product of what this document is about. 
```

I think "(Replace this text with your content)" is more confusing here than helpful.

File: source/documentors/references/templates/doc_reference_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_reference_template.rst:9]:
```suggestion
For example, list details or a product glossary that the reader could need to use.
```
I think "(Replace this text with your content)" is more confusing here than helpful.

File: source/documentors/references/templates/doc_reference_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_reference_template.rst:57]:
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_reference_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_reference_template.rst:127]:
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:54]:
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:103]:
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:222]:
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_how_to_template.rst:35]:
```suggestion
Now, you have a how-to document. Congratulations! 
```

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_how_to_template.rst:79]:
```suggestion
	Now, you have a how-to document. Congratulations! 
```

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:1]:
```suggestion
Quickstart Document Template
```
Be sure you're using an English spellchecker on these documents

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:18]:
```suggestion
A good quickstart is designed for users who want to dive in with minimal setup. For example, allowing developers to try out a tool or platform without wading through extensive documentation upfront, or empowering course authors to quickly get up and running with a complex area of Studio.
```

To call out that Quickstarts are not _just_ for developers, which is how this currently reads. Quickstarts are for all personas.

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:78]:
```suggestion
Here is a list of the most used resources to create a Quickstart:
```

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [reply source/documentors/references/templates/doc_quickstart_template.rst:112]: Yes, please link to the Style Guide

Comment (sarina) [reply source/documentors/references/templates/doc_quickstart_template.rst:112]: And please add this See Also to all 4 template guides.

**Review by sarina — CHANGES_REQUESTED** [review]

File: source/documentors/references/templates/doc_concept_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_concept_template.rst:6]: My comment about the image example was that this line ("What is an XBlock") might be clearer if the associated image _also_ was about the same thing. So you changed the image to one about Superset, which is still unrelated to this line about XBlocks. It's not a big deal but that was what I was saying.

File: source/documentors/references/templates/doc_quickstart_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_quickstart_template.rst:115]: This isn't rendering in the output

image

https://docsopenedxorg--619.org.readthedocs.build/en/619/documentors/references/templates/doc_quickstart_template.html

File: source/documentors/references/templates/doc_reference_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_reference_template.rst:69]: not rendering
image

File: source/documentors/references/templates/doc_concept_template.rst
Comment (Apgomeznext) [reply source/documentors/references/templates/doc_concept_template.rst:6]: You are right. Sorry about the confusion.

**Comment by Apgomeznext** [comment]
The only thing that needs to be added to these documents is the link to the style guide. The error that this PR showed this morning was because I had added the link as it should have once it was published, but I deleted it to avoid the error.

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_how_to_template.rst:31]:
```suggestion
.. seealso:: To learn more about how to add a code block, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```

File: source/documentors/references/templates/doc_how_to_template.rst
Comment (sarina) [inline source/documentors/references/templates/doc_how_to_template.rst:77]:
```suggestion
	.. seealso:: To learn more about how to add a code block, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```

**Review by sarina — APPROVED** [review]

**Pull request merged**
